packages/flutter_tools/lib/src/project.dart

```diff
@@ -581,13 +586,17 @@ the v1 Android embedding will be removed in future versions of Flutter.
   }
 
   AndroidEmbeddingVersion getEmbeddingVersion() {
+    computeEmbeddingVersion().version;
+  }
+
+  AndroidEmbeddingVersionResult computeEmbeddingVersion() {
     if (isModule) {
       // A module type's Android project is used in add-to-app scenarios and
       // only supports the V2 embedding.
-      return AndroidEmbeddingVersion.v2;
+      return AndroidEmbeddingVersionResult(AndroidEmbeddingVersion.v2, 'Is add-to-app module');
     }
     if (appManifestFile == null || !appManifestFile.existsSync()) {
-      return AndroidEmbeddingVersion.v1;
+      return AndroidEmbeddingVersionResult(AndroidEmbeddingVersion.v1, 'No "android/app/src/main/AndroidManifest.xml" file');
```

Comment:
nit: `appManifestFile.absolute.path`.

Some scripts like the ones in the plugins repo, run a bunch of commands in different directories, so it's not straightforward to know which AndroidManifest.xml this is referring to.